pandas/io/parsers.py

```diff
@@ -997,14 +997,14 @@ def _validate_usecols_arg(usecols):
 
     if usecols is not None:
         if callable(usecols):
-            return usecols
+            return usecols, None
         usecols_dtype = lib.infer_dtype(usecols)
         if usecols_dtype not in ('empty', 'integer',
                                  'string', 'unicode'):
             raise ValueError(msg)
 
-        return set(usecols)
-    return usecols
+        return set(usecols), usecols_dtype
```

Comment:
@jreback : I updated the documentation to illustrate the return type is independent of engine.